sdk/storage/azdatalake/internal/shared/challenge_policy.go

```diff
@@ -0,0 +1,113 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package shared
+
+import (
+	"errors"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"net/http"
+	"strings"
+)
+
+type storageAuthorizer struct {
```

Comment:
Do we have a challenge policy in other SDKs too? This was added to blob specifically for managed disks, so I dont think it is relevant to HNS accounts. Thoughts?

Reply:
I assumed if it was under shared in blob it applied to the other packages. Haven't looked closely at this yet. Can be removed if this is the case.